pkg/scalers/openstack_swift_scaler.go

```diff
@@ -0,0 +1,374 @@
+		isOnlyFiles, conversionError := strconv.ParseBool(val)
+		if conversionError != nil {
+			return nil, fmt.Errorf("onlyFiles parsing error: %s", conversionError.Error())
+		}
+		meta.onlyFiles = isOnlyFiles
+	} else {
+		meta.onlyFiles = defaultOnlyFiles
+	}
+
+	if val, ok := config.TriggerMetadata["objectLimit"]; ok {
+		meta.objectLimit = val
+	} else {
+		meta.objectLimit = defaultObjectLimit
+	}
+
+	return &meta, nil
+}
+
+func parseSwiftAuthenticationMetadata(config *ScalerConfig) (*swiftAuthenticationMetadata, error) {
+	authMeta := swiftAuthenticationMetadata{}
+
+	if config.AuthParams["authURL"] != "" {
+		authMeta.authURL = config.AuthParams["authURL"]
+	} else {
+		return nil, fmt.Errorf("authURL doesn't exist in the authParams")
+	}
+
+	if config.AuthParams["userID"] != "" {
+		authMeta.userID = config.AuthParams["userID"]
+
+		if config.AuthParams["password"] != "" {
+			authMeta.password = config.AuthParams["password"]
+		} else {
+			return nil, fmt.Errorf("password doesn't exist in the authParams")
+		}
+
+		if config.AuthParams["projectID"] != "" {
+			authMeta.projectID = config.AuthParams["projectID"]
+		} else {
+			return nil, fmt.Errorf("projectID doesn't exist in the authParams")
+		}
+	} else {
+		if config.AuthParams["appCredentialID"] != "" {
+			authMeta.appCredentialID = config.AuthParams["appCredentialID"]
+
+			if config.AuthParams["appCredentialSecret"] != "" {
+				authMeta.appCredentialSecret = config.AuthParams["appCredentialSecret"]
+			} else {
+				return nil, fmt.Errorf("appCredentialSecret doesn't exist in the authParams")
+			}
+		} else {
+			return nil, fmt.Errorf("neither userID or appCredentialID exist in the authParams")
+		}
+	}
+
+	return &authMeta, nil
+}
+
+func (s *swiftScaler) IsActive(ctx context.Context) (bool, error) {
+	objectCount, err := s.getSwiftContainerObjectCount()
+
+	if err != nil {
+		return false, err
+	}
+
+	return objectCount > 0, nil
+}
+
+func (s *swiftScaler) Close() error {
+	return nil
+}
+
+func (s *swiftScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	objectCount, err := s.getSwiftContainerObjectCount()
+
+	if err != nil {
+		swiftLog.Error(err, "error getting objectCount")
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(objectCount), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *swiftScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetObjectCount := resource.NewQuantity(int64(s.metadata.objectCount), resource.DecimalSI)
+
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s", "swift", s.metadata.containerName)),
```

Comment:
The generated `metricName` should be unique enough, that if there are multiple scalers of the same type in one ScaledObject, each metric name is still unique.
Just double checking, because I don't have enough expertise in Openstack Swift, I am not sure whether this is the case? 

Do you think that there's a sane scenario where we would like to have multiple swift triggers pointing to the same container? Or is that something, that doesn't make a sense?

Reply:
OpenStack Swift should work similarly to Azure Blob Storage or any other object-store service.

We have identified one possible scenario where it could be possible to have multiple swift triggers pointing to the same container, though: when one wants to scale based on the number of objects inside different folders inside the same container. In this case, delimiter and prefixes can be used. In order to set more than one prefix, new triggers for the same container in the same ScaledObject must be created.

Thus, we have modified `metricName` to contain the `containerName` + `objectPrefix`, when the latter is provided. That should be enough to cover this scenario.

Reply:
Sounds good, thanks!